pkg/servicelb/controller.go

```diff
@@ -553,7 +549,7 @@ func (h *handler) updateDaemonSets() error {
 // deleteOldDeployments ensures that there are no legacy Deployments for ServiceLB pods.
 // ServiceLB used to use Deployments before switching to DaemonSets in 875ba28
 func (h *handler) deleteOldDeployments(svc *core.Service) error {
-	name := fmt.Sprintf("svclb-%s", svc.Name)
+	name := fmt.Sprintf("svclb-%s-%s", svc.Name, svc.UID[:8])
```

Comment:
I'm not sure we need this change, since any (very very) legacy Deployments wouldn't be generated using the new naming scheme.

Reply:
You're right, updated